src/nemos/basis.py

```diff
@@ -223,6 +236,21 @@ def transform(self, X: FeatureMatrix, y=None) -> FeatureMatrix:
         -------
         :
             The data transformed by the basis functions.
+
+        Examples
+        --------
+        # Example input
+        >>> import numpy as np
+        >>> X, y = np.random.normal(size=(100, 2)), np.random.uniform(size=100)
+
+        # Define and fit tranformation basis
+        >>> from nemos.basis import MSplineBasis, TransformerBasis
+        >>> basis = MSplineBasis(10)
+        >>> transformer = TransformerBasis(basis)
+        >>> transformer_fitted = transformer.fit(X) # input must be a 2d array
```

Comment:
I would comment about what "fit" is doing here. Say something like: fit in mode"conv" creates the set of basis that will be convolve with the input. In mode "eval" does nothing, but we need the method to fully comply with sklearn API.

Reply:
Assume that the
```suggestion
        >>> # Before calling `fit` the convolution kernel is not set
        >>> transformer.kernel_
        
        >>> transformer_fitted = transformer.fit(X) # input must be a 2d array
        >>> # Now the convolution kernel is initialized and has shape (window_size, n_basis_funcs)
        >>> transformer_fitted.kernel.shape
        (200, 10)
        
```